### ISSUE TITLE
Add control with inline math rendering capabilities

### ISSUE THREAD

**Issue by illiteratePhD** [ForNeVeR/wpf-math#77]
Current existing custom control would be cumbersome to use to produce paragraphs of text with inline math or longer flowing text with math as separate lines. See example below

http://marked2app.com/help/images/mathjax2.jpg

A control based on a textbox, but with added handles such as [$, $] that uses the renderer would be a very useful addition.

**Comment by ForNeVeR** [comment]
Yes, that's right. Today WPF-Math doesn't help much with formatting whole TeX documents within WPF forms or documents.

I believe that `FlowDocument` is the way to go, but I'd like someone (including myself) to conduct more research on that, and maybe even implement some simple prototype. I still don't know what changes that'll require, and we need to ascertain that before doing anything.

**Comment by alexreg** [comment]
I think this would be beyond the scope of the project. It would be great if someone implements this, but I think it's a big project, and probably best we stay clear of it for now.

**Comment by illiteratePhD** [comment]
I've looked at the FlowDocument tag a bit and it turns out adding WPF-Math is really not that hard, if you can accept not being able to use the standard $...$ pattern. By changing the `<border>` section of the example with the following:

```
<Border Margin="10,0,10,10" BorderBrush="LightGray" BorderThickness="1" Grid.Row="3" Grid.Column="0">
    <FlowDocumentScrollViewer>
        <FlowDocument>
            <Paragraph>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce faucibus odio arcu, luctus vestibulum tortor congue in. Lorem ipsum dolor sit amet, consectetur,
                <InlineUIContainer BaselineAlignment="TextBottom">
                    <controls:FormulaControl x:Name="formula1" Formula="\sum d\theta = \tan^{-1}\theta+C" Scale="12" SnapsToDevicePixels="True"/>
                </InlineUIContainer>
                adipiscing elit. Fusce nec lacinia neque. Donec malesuada, ligula non vestibulum cursus, urna purus pellentesque orci, 
                aliquet accumsan dui velit ac justo. 
            </Paragraph>
            <BlockUIContainer>
                <controls:FormulaControl x:Name="formula" Formula="L = \int_a^b \sqrt[4]{ \left| \sum_{i,j=1}^ng_{ij}\left(\gamma(t)\right) \left[\frac{d}{dt}x^i\circ\gamma(t) \right] \left{\frac{d}{dt}x^j\circ\gamma(t) \right} \right|}dt" Scale="12" SnapsToDevicePixels="True"/>
            </BlockUIContainer>
            <Paragraph>
                Nulla vitae suscipit tellus. Nunc sit amet tortor fermentum, sollicitudin enim cursus, sagittis lacus. Pellentesque tincidunt massa nisl, nec tempor nulla 
                consequat a. Proin pharetra neque vel dolor congue, at condimentum arcu varius. Sed vel luctus enim. Curabitur eleifend dui et arcu faucibus, sit amet 
                vulputate libero suscipit. 
            </Paragraph>
        </FlowDocument>
    </FlowDocumentScrollViewer>
</Border>
```
You actually get something that looks decent. The problem with the inline container is that the baseline, of course, is off. So essentially, what's needed, is a way to switch between the normal and in-line rendering of the formula.

Loving this project. Wish there was a way I could contribute in a meaningful way. Trying to get into the code, but it's a slow process :)

**Comment by alexreg** [comment]
Okay, well maybe we could created a *contrib* repository and put your control there, if you get it working a bit more smoothly. :) It's been a long time since I did anything with flow documents, but that's good you've made some initial success.

**Comment by illiteratePhD** [comment]
On the topic of inline vs. display. How are these things handled by existing TeX parsers? Are there two character sets or is the existing character set adjusted to a different environment?

**Comment by ForNeVeR** [comment]
@alexreg I don't think we need to split our project into more fragments at the moment. If we can have a decent and *non-intrusive* solution to use WPF-Math in `<FlowDocument>` context, I'll vote to accept such contribution and include in into the main code base.

_Side note: I'll be absent from March 23rd to 24th on a small scientific trip; I hope you'll be fine here guys 😄_

**Comment by alexreg** [comment]
Hmm, well it's your decision @ForNeVeR, but I think it should at the *least* be in a separate VSproj, if not a separate repository.

**Comment by B3zaleel** [comment]
I think I managed to create a control that can render the formula control with normal text, but a lot of the features are still under development. I Would push it when it is quite decent.

**Comment by ForNeVeR** [comment]
I would very much ask you to please create a separate branch for that.

**Comment by B3zaleel** [comment]
I will
